pkg/deploy/api/validation/validation.go

```diff
@@ -154,6 +154,14 @@ func validateLifecycleHook(hook *deployapi.LifecycleHook, pod *kapi.PodSpec, fld
 		errs = append(errs, field.Required(fldPath.Child("failurePolicy"), ""))
 	}
 
+	if hook.MaxRetries > 0 && hook.FailurePolicy != deployapi.LifecycleHookFailurePolicyRetry {
+		errs = append(errs, field.Invalid(fldPath, "<hook>", "maxRetries can be used only with the 'Retry' failurePolicy"))
+	}
+
+	if hook.MaxRetries < 0 && hook.FailurePolicy == deployapi.LifecycleHookFailurePolicyRetry {
```

Comment:
i see, good catch. maxRetries is only allowed for "retry" policy, so if it
is set to smoething else than 0 for other policies, we should refuse that.

On Fri, Mar 4, 2016 at 4:30 PM, David Eads <notifications@github.com> wrote:

> In pkg/deploy/api/validation/validation.go
> <https://github.com/openshift/origin/pull/7767#discussion_r55043384>:
>
> > @@ -154,6 +154,14 @@ func validateLifecycleHook(hook *deployapi.LifecycleHook, pod *kapi.PodSpec, fld
> >  		errs = append(errs, field.Required(fldPath.Child("failurePolicy"), ""))
> >  	}
> >
> > +	if hook.MaxRetries > 0 && hook.FailurePolicy != deployapi.LifecycleHookFailurePolicyRetry {
> > +		errs = append(errs, field.Invalid(fldPath, "<hook>", "maxRetries can be used only with the 'Retry' failurePolicy"))
> > +	}
> > +
> > +	if hook.MaxRetries < 0 && hook.FailurePolicy == deployapi.LifecycleHookFailurePolicyRetry {
>
> Why would you ever allow negative values?
>
> maxRetries == -1 and FailurePolicy == LifecycleHookFailurePolicyAbort ?
>
> I would have expected
>
> if maxRetries < 0{
>     error
> }
>
> Why is there another predicate?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7767/files#r55043384>.
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering